LIBFCREPO-1344. Added VS Code Dev Container setup and documentation

dsteelma-umd opened the pull request:
* Added a “Dockerfile.vscode” file for creating a suitable Docker image for local development, including the addition of the “xmlsec1” system package needed for SAML

* Added “.devcontainer/devcontainer.json” for configuring the VS Code Dev Container, including providing the following VS Code extensions:
    * Python (ms-python.python)
    * Ruff (charliermarsh.ruff)
    * markdownlint (davidanson.vscode-markdownlint)

* Modified the “README.md”, replacing the existing local development instructions with two separate files:
    * docs/DevelopmentEnvironmentLocal.md
    * docs/DevelopmentEnvironmentVsCode.md

* Updated the “.gitignore” file to ignore the SAML key and certificate files

https://umd-dit.atlassian.net/browse/LIBFCREPO-1344

dsteelma-umd committed:
LIBFCREPO-1344. Added VS Code Dev Container setup and documentation

* Added a “Dockerfile.vscode” file for creating a suitable Docker image
  for local development, including the addition of the “xmlsec1” system
  package needed for SAML. The "vim" editor was also included for
  use when writing Git commits.

* Added “.devcontainer/devcontainer.json” for configuring the VS Code
  Dev Container, including providing the following VS Code extensions:

    * Python (ms-python.python)
    * Ruff (charliermarsh.ruff)
    * markdownlint (davidanson.vscode-markdownlint)
    * GitLens (eamodio.gitlens)

* Modified the “README.md”, replacing the existing local development
  instructions with two separate files:

    * docs/DevelopmentEnvironmentLocal.md
    * docs/DevelopmentEnvironmentVsCode.md

* Updated the “.gitignore” file to ignore the SAML key and certificate
files

* Added an "env_example" file to provide a template for the ".env"
  file

https://umd-dit.atlassian.net/browse/LIBFCREPO-1344

peichman-umd commented on .gitignore:
```diff
@@ -160,4 +160,12 @@ cython_debug/
 public
+
+# SAML key/certificate files
+**/grove*.cer
```
The extension used on the certificate in LastPass is `.crt`

peichman-umd commented on docs/DevelopmentEnvironmentLocal.md:
````diff
@@ -0,0 +1,128 @@
+    ```
+
+3) Download the `grove-test-lib-umd-edu-sp.key` and
+  `grove-test-lib-umd-edu-sp.cer` files from the  "grove-local-saml" entry in
````
The extension used on the certificate in LastPass is `.crt`

peichman-umd commented on docs/DevelopmentEnvironmentLocal.md:
```diff
@@ -0,0 +1,128 @@
+    * SAML_KEY_FILE - relative (or absolute) file path to the
+      `grove-test-lib-umd-edu-sp.key` file
+    * SAML_CERT_FILE - relative (or absolute) file path to the
+      ``grove-test-lib-umd-edu-sp.cer` file
```
The extension used on the certificate in LastPass is `.crt`

peichman-umd commented on docs/DevelopmentEnvironmentVsCode.md:
```diff
@@ -0,0 +1,113 @@
+   The dev container will build the Docker image (if necessary), and install
+   Python dependencies.
+
+3) Download the `grove-test-lib-umd-edu-sp.cer` and
```
The extension used on the certificate in LastPass is `.crt`

peichman-umd commented on docs/DevelopmentEnvironmentVsCode.md:
````diff
@@ -0,0 +1,113 @@
+    ```zsh
+    cp env_example .env &&
+       sed -i '/SAML_KEY_FILE=/cSAML_KEY_FILE=grove-test-lib-umd-edu-sp.key' .env &&
+       sed -i '/SAML_CERT_FILE=/cSAML_CERT_FILE=grove-test-lib-umd-edu-sp.cer' .env &&
````
The extension used on the certificate in LastPass is `.crt`

peichman-umd reviewed (changes requested):
The SAML certificate store in LastPass uses `.crt` as its extension, so the documentation and .gitignore should be updated to reflect that.

dsteelma-umd committed:
LIBFCREPO-1344. Added note about downloading certifcate file in Chrome

When downloading the "grove-test-lib-umd-edu-sp.crt" file from LastPass,
Google Chrome changes the file extension, by defauly, from ".crt" to
".cer". Added a note about this to the development environment
instructions, indicating that ".crt" should be used.

Mozilla Firefox preserves the ".crt" extension.

Also added the `**/grove*.crt` to the ".gitignore" file, so both
certificate file extensions will be ignored by Git.

https://umd-dit.atlassian.net/browse/LIBFCREPO-1344
